src/DotNetWorker/Definition/FunctionParameter.cs

```diff
@@ -5,19 +5,30 @@
 
 namespace Microsoft.Azure.Functions.Worker.Definition
 {
+    /// <summary>
+    /// Represents a parameter defined by the target function.
+    /// </summary>
     public class FunctionParameter
     {
+        /// <summary>
+        /// Creates an instance of the <see cref="FunctionParameter"/> class.
+        /// </summary>
+        /// <param name="name">The parameter name.</param>
+        /// <param name="type">The <see cref="System.Type"/> of the parameter.</param>
         public FunctionParameter(string name, Type type)
         {
-            Name = name;
-            Type = type;
+            Name = name ?? throw new ArgumentNullException(nameof(name));
+            Type = type ?? throw new ArgumentNullException(nameof(type));
         }
 
+        /// <summary>
+        /// Gets the parameter name.
+        /// </summary>
         public string Name { get; }
 
+        /// <summary>
+        /// Gets the parameter <see cref="System.Type"/>.
+        /// </summary>
         public Type Type { get; }
-
-        // TODO: Pop out to Context (or Invocation)
-        public object? Value { get; set; }
     }
```

Comment:
Do you want to add Properties to this now? I just remembered one use for this -- we can stick the GrpcType in here (i.e. JSON/Byte[]) so conversion is quicker.

Reply:
Would this require the converters to be aware of that type? Or are we planning on having gRPC specific converters (which would make sense)?

Reply:
Adding the properties. Will just have that dictionary to begin with, we can talk about potential optimizations in the future